next-app/src/components/common/Layout/Layout.tsx

```diff
@@ -11,13 +11,14 @@ interface Props {
 
 const Layout = ({ children }: Props) => {
   const router = useRouter()
-  const isSignUpPage = useMemo(() => router.pathname === '/signup', [router])
-  const isOauthPage = useMemo(() => router.pathname === '/oauth', [router])
+  const isRequiredAuthPage = requiredAuthMatcher(router.pathname)
+  const isSignUpPage = router.pathname === '/signup'
+  const isIntroducePage = router.pathname === '/introduce'
 
   return (
     <>
-      {!(isSignUpPage || isOauthPage) && <Nav />}
-      <Container isSignUpPage={isSignUpPage}>
+      {(isRequiredAuthPage || isIntroducePage) && <Nav />}
+      <Container fullHeight={isSignUpPage}>
```

Comment:
결과적으로 보면 Nav가 없는 페이지에서는 fullHeight가 되어야 하는데, 서로 같은 변수를 이용해서 컨트롤 하는게 더 좋지 않을까 하는 생각이 드네요. 그리고 지금은 Nav가 필요없는 페이지가 isSignUp 페이지 이지만 추후에 Nav가 필요없는 페이지가 추가될 가능성에 따라서 ignoreNavPage를 따로 관리하면 어떨까 싶습니다
```suggestion
      {!isIgnoreNavPage && <Nav />}
      <Container fullHeight={isIgnoreNavPage}>
```

Reply:
@eunsonny 아래와 같이 반영했습니다

```tsx
const Layout = ({ children }: Props) => {
  const router = useRouter()

  const { isErrorPage, isIntroducePage, isRequiredAuthPage } = routerBranch(
    router.pathname
  )
  const appearGNB = isRequiredAuthPage || isIntroducePage

  return (
    <>
      {appearGNB && <Nav />}
      <Container fullHeight={!appearGNB || isErrorPage}>
        <main>{children}</main>
      </Container>
    </>
  )
}

export default Layout

const routerBranch = (pathname: string) => {
  return {
    isRequiredAuthPage: requiredAuthMatcher(pathname),
    isIntroducePage: pathname === '/introduce',
    isErrorPage: isErrorPage(pathname),
  }
}

```